src/applications/model/four-elements-streaming-server.cc

```diff
@@ -45,14 +45,14 @@ FourElementsStreamingServer::GetTypeId (void)
 FourElementsStreamingServer::FourElementsStreamingServer ()
```

Comment:
In this case, I need to call the complete constructor by setting default values for Address IP and port (by random or fixed values). Should I do this?

I think, In constructor delegation, it works inversely: Having a base (simple) constructor, we call it within the complete constructor(s) to set basic attributes.
Here because I'm calling the Base-Class constructor (`GamingStreamingServer`), I can't call its own simple constructor.

Reply:
If you call the default constructor from the complete constructor you have to initialize variables to default values then set the correct values.
If you call the complete constructor from the default one you can initialize variables by passing default values.

Both work for me, maybe check what is the best practice when using delegating ctors.

I don't get the last sentence, though.

Reply:
My last sentence: Here the `FourElementStreamingServer` Derived from `GamingStreamingServer`, and it's not possible to call the **default constructor of a child class** and the **constructor of the parent class**, in the **complete constructor of the child class** at the same time.

I'm going to do this:
> If you call the complete constructor from the default one you can initialize variables by passing default values.

but the question is how should I choose the default values for Address and port? by random or I should assign them a fixed value?

Reply:
```
FourElementsStreamingServer::FourElementsStreamingServer ()
  : FourElementsStreamingServer (Address (), 21)
{
  NS_LOG_FUNCTION (this);
}

This is what I'm talking about. I'm not sure calling complete constructor for the default one is a good idea.